### PR TITLE
Win_schedule_tasks strict mode fixes, periodicity updates, working directory

### PR DESCRIPTION
**Pull request by perzizzle**
##### ISSUE TYPE
- Feature Pull Request
##### COMPONENT NAME

win_scheduled_task
##### ANSIBLE VERSION

```
ansible 2.0.1.0
```
##### SUMMARY

This PR provides:
- additional strict mode fixes (use get-attr to get $task.state to handle the initial creation of a scheduled task)
- removes unneeded code for getting the $path attribute, specifies default value
- adds support for specifying minutely, hourly periods
- adds support for specifying a working directory

### CODE REVIEW

**Comment by perzizzle** [comment]
@petemounce I made some additional fixes to support ansible 2.0 and merged in some features I've been using locally. Is there a reason I'm not a contributor for this module? I did the initial re-write of the module, I think I'm trustworthy ;)

**Comment by petemounce** [comment]
No reason at all! Add yourself :)

##

**Comment by gregdek** [comment]
Thanks @perzizzle. To the current maintainers, @petemounce please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit', 'needs_revision' or 'close_me' as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
Thanks @perzizzle for this PR. This PR requires revisions, either because it fails to build or by reviewer request. Please make the suggested revisions. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by petemounce** [comment]
lgtm. I'm going to say needs_revision only so that @perzizzle has the opportunity to edit the authors list within the same PR.

**Comment by perzizzle** [comment]
@gregdek Looks like the build failed because it doesn't like the version_added attribute in the documentation. I couldn't find any guidance at http://docs.ansible.com/ansible/developing_modules.html on how to define this string. Currently, its set to 2.0 (although the module existed before 2.0).

@petemounce what do you want for your name in the author string? I can add it in this PR if you like. Peter Mounce public@neverrunwithscissors.com
Michael Perzel michaelperzel@gmail.com
is currently in the ps1 file.

**Comment by gregdek** [comment]
@perzizzle The issue is that you're adding new functions in the description field. New functions must be accompanied by the version that those features were added, by adding a version_added field to the description for that feature. In this case, it would be 2.1.

**Comment by perzizzle** [comment]
@petemounce @gregdek ready_for_review

**Comment by petemounce** [comment]
shipit! Thanks :)

**Comment by gregdek** [comment]
Thanks again to @perzizzle for this PR, and thanks @petemounce for reviewing. Marking for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

File: windows/win_scheduled_task.py [hunk 1]
```diff
@@ -67,6 +70,8 @@
       - The frequency of the command, not idempotent
     required: false
     choices:
+      - minutely
+      - hourly
```
Comment (resmo) [inline windows/win_scheduled_task.py:74]: @perzizzle please make a note in the description, that the new choices `minutely`, `hourly` were added added in 2.2.

Comment (alikins) [inline windows/win_scheduled_task.py:74]: Flipped back to needs_revision for this.

**Comment by gregdek** [comment]
@perzizzle A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes. When you do address the issues, please respond with ready_for_review in your comment, so that we can notify the maintainer.

[This message brought to you by your friendly Ansibull-bot.]

File: windows/win_scheduled_task.py (hunk 1 above)
Comment (perzizzle) [inline windows/win_scheduled_task.py:74]: @alikins Added a note about minutely, hourly being added

**Comment by perzizzle** [comment]
@alikins Added more info in description about features added. The build failing doesn't seem related to anything I touched.

**Comment by gregdek** [comment]
@perzizzle Another friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open. If you have addressed the issues and believe it's ready for review, please comment with the text "ready_for_review". If we don't hear from you within another 14 days, we will close this pull request.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@perzizzle A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes. When you do address the issues, please respond with ready_for_review in your comment, so that we can notify the maintainer.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by perzizzle** [comment]
@gregdek This was passing checks before and the failures seem to be unrelated to anything I've done. I think this is ready to go. Its had a couple shipits.

**Comment by gregdek** [comment]
@perzizzle Another friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open. If you have addressed the issues and believe it's ready for review, please comment with the text "ready_for_review". If we don't hear from you within another 14 days, we will close this pull request.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@perzizzle just rebase against latest and it should be good to go.

**Comment by dreamcat4** [comment]
Hi there. Was this merged yet? I'm on git today and getting `The property 'State' cannot be found on this object.`

**Comment by dreamcat4** [comment]
Oh OK no worries. I check the github in `devel` right now and its simply not been merged yet.

https://github.com/ansible/ansible-modules-extras/blob/devel/windows/win_scheduled_task.ps1

Sorry just confusing for me why PR is closed for 6 days ago if its not in upsteam git yet. Maybe you had a difficulty rebasing on head etc with this particular one?

**Comment by perzizzle** [comment]
@dreamcat4 This PR took so long to get merged that commit eed9d601b5b727586e16f3afa5824031dfbc88df included some of the fixes I was doing for Strict Mode so I closed it. I could open a new PR for setting the working directory if people are interested.

**Comment by dreamcat4** [comment]
Hi @perzizzle... Just want to do a playbook run without it erroring out. `The property 'State' cannot be found on this object.` Which it seems is not in the `devel` branch yet on `ansible/ansible-modules-extras` ?

**Comment by dreamcat4** [comment]
Oh! Maybe you did merge it already, Just that I had not pulled in my other git clone local `libraries/` repo folder (which overrides the other one). I shall just try again now.

**Comment by dreamcat4** [comment]
Yeah OK sorry. This works for me now. Due to your earlier / other commit. Error message has gone away. Many thanks.

**Pull request closed without merging**